app/src/org/commcare/dalvik/activities/CrashWarningActivity.java

```diff
@@ -0,0 +1,79 @@
+package org.commcare.dalvik.activities;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.graphics.Color;
+import android.os.Bundle;
+import android.text.Spannable;
+import android.text.SpannableStringBuilder;
+import android.text.style.ForegroundColorSpan;
+import android.text.style.StyleSpan;
+import android.view.View;
+import android.widget.Button;
+import android.widget.TextView;
+
+import org.commcare.android.util.CommCareExceptionHandler;
+import org.commcare.dalvik.R;
+import org.javarosa.core.services.locale.Localization;
+
+/**
+ * Shows reason for unrecoverable crash to user.
+ */
+public class CrashWarningActivity extends Activity {
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        setContentView(R.layout.activity_crash_warning);
+
+        Button closeButton = (Button)findViewById(R.id.CloseWarning);
+        closeButton.setText(Localization.get("crash.warning.button"));
+        closeButton.setOnClickListener(new View.OnClickListener() {
+            public void onClick(View v) {
+                setResult(RESULT_OK);
+                finish();
+            }
+        });
+
+        Intent intent = getIntent();
+        setWarningText(intent);
```

Comment:
Will these methods ever be performed separately/one and not the other? If not maybe makes sense to combine to one method?